src/steps/index.test.ts

```diff
@@ -133,3 +133,140 @@ test('should collect data', async () => {
     },
   });
 });
+
+test('stopFetchingRequests', () => {
```

Comment:
I think it would be good to break these expectations up into separate tests, something like:

```
describe('stopFetchingRequests', () => {
  test('lastExecutionTime undefined ingests max age')
  test('lastExecutionTime 0 ingests max age')
  test('requests missing updateAt ingests limited number')
  test('requests in first page are really old, stops paginating')
  ...
});

Reply:
Great, thank you, that changed my thought process on tests into something a lot more sophisticated.